server/src/main/java/org/opensearch/common/logging/DeprecatedMessage.java

```diff
@@ -47,12 +47,17 @@
  */
 public class DeprecatedMessage extends OpenSearchLogMessage {
     public static final String X_OPAQUE_ID_FIELD_NAME = "x-opaque-id";
-    private static final Set<String> keys = ConcurrentHashMap.newKeySet();
+
+    // Arbitrary maximum size, should be much larger than unique number of
+    // loggers, but small relative to heap size.
+    static final int MAX_DEDUPE_CACHE_ENTRIES = 65_536;
```

Comment:
why do we need 64k keys for deprecation logger? can we reduce it further to lets say 8k or 16k only? why would user have so many clients?

Reply:
I honestly don't know the right number. I just picked a power of 2 :) I can reduce it.

Reply:
yeah lets reduce it. same here, i also dont have full context how x-opaque-id is used by users outside of opensearch-clients.